core/arch/arm/mm/core_mmu.c

```diff
@@ -222,6 +226,87 @@ static struct tee_mmap_region *find_map_by_pa(unsigned long pa)
 	return NULL;
 }
 
+extern const struct core_mmu_phys_mem __start_phys_sdp_mem_section;
+extern const struct core_mmu_phys_mem __end_phys_sdp_mem_section;
+
+static bool pbuf_is_sdp_mem(paddr_t pbuf, size_t len)
+{
+	const struct core_mmu_phys_mem *mem;
+
+	for (mem = &__start_phys_sdp_mem_section;
+	     mem < &__end_phys_sdp_mem_section; mem++)
+		if (core_is_buffer_inside(pbuf, len, mem->addr, mem->size))
+			return true;
+
+	return false;
+}
+
+static bool pbuf_intersect_sdp_mem(paddr_t pbuf, size_t len)
+{
+	const struct core_mmu_phys_mem *mem;
+
+	for (mem = &__start_phys_sdp_mem_section;
+	     mem < &__end_phys_sdp_mem_section; mem++)
+		if (core_is_buffer_intersect(pbuf, len, mem->addr, mem->size))
+			return true;
+
+	return false;
+}
+
+static bool verify_sdp_mem_areas(void)
+{
+	const struct core_mmu_phys_mem *mem1;
+	const struct core_mmu_phys_mem *mem2;
+	const struct core_mmu_phys_mem *mem_start = &__start_phys_sdp_mem_section;
+	const struct core_mmu_phys_mem *mem_end = &__end_phys_sdp_mem_section;
+
+	for (mem1 = mem_start; mem1 < mem_end; mem1++)
+		for (mem2 = mem1 + 1; mem2 < mem_end; mem2++)
+			if (core_is_buffer_intersect(mem2->addr, mem2->size,
+						     mem1->addr, mem1->size))
+				goto err;
+
+	return true;
+err:
+	EMSG("[%" PRIxPA " %" PRIxPA "] intersecs [%" PRIxPA " %" PRIxPA "]",
+			mem2->addr, mem2->addr + mem2->size,
+			mem1->addr, mem1->addr + mem1->size);
+	return false;
+
+}
+
+struct mobj **core_sdp_mem_create_mobjs(void)
+{
+	const struct core_mmu_phys_mem *mem;
+	struct mobj **mobj_base;
+	struct mobj **mobj;
+	int cnt;
+
+	if (!verify_sdp_mem_areas())
+		panic("Invalid SDP memory areas");
+
+	for (mem = &__start_phys_sdp_mem_section, cnt = 0;
+	     mem < &__end_phys_sdp_mem_section; mem++, cnt++)
+		;
+
+	mobj_base = calloc(cnt + 1, sizeof(*mobj));
+	if (!mobj_base)
+		panic("oom");
```

Comment:
Oom? Did you mean boom :boom: ? :) ... anyhow, I think the full "Out of Memory" or something like that would be better.